src/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+# ===========================================
+# Set up navier-stokes library
+
+add_library(navier_stokes INTERFACE "navier_stokes/navier_stokes.h")
+add_library(feltor::navier_stokes ALIAS navier_stokes)
+target_link_libraries(navier_stokes INTERFACE dg::dg)
+
+# ============================================
+# Set up FELTOR
+# TODO: Other executables
+
+set(FELTOR_HEADERS
+    "feltor/feltor.h"
+    "feltor/init_from_file.h"
+    "feltor/init.h"
+    "feltor/parameters.h"
+)
+
+add_executable(feltor "feltor/feltor.cpp" ${FELTOR_HEADERS})
+add_executable(feltor::feltor ALIAS feltor)
+
+target_include_directories(feltor PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+
+if(FELTOR_USE_MPI)
+    target_link_libraries(feltor PRIVATE MPI::MPI_CXX)
+    target_compile_definitions(feltor PRIVATE WITH_MPI)
+endif()
+
+# TODO: Replace following commands with portable alternatives.
```

Comment:
Do these commands not work on windows or Mac?

Reply:
I would expect the `git` commands to work in all three (though I'm not 100% sure about Windows), and I believe the `date` command is the same on Mac, but the Windows `date` command is different.

There is a platform independent way to get a timestamp in CMake (see these [StackOverflow answers](https://stackoverflow.com/questions/24292898/compile-date-and-time-using-cmake)), but I'm not sure if you can replicate the exact same format. I'll look into it.